src/modules/dirwatcher.js

```diff
@@ -0,0 +1,37 @@
+import EventEmitter from 'events'
+import fs from 'fs'
+import Set from 'collections/set'
```

Comment:
Avoid using `collections` package. I understand, that you want to use relative to you structural types, but there're native `Array, Set & Map` types, and Array is used mostly always